### DISCUSSION TITLE
Single screenshot in allure reports for multiple capabilities.

### DISCUSSION THREAD

**Discussion by VarunJain1402** [webdriverio/webdriverio#13155] [Q&A]
I have used four different capabilites in my tests and added step to take screenshot in the aftertest in mocha. So, I want to have screenshots to appear in my report for as many capability i am defining. The issue is only one screenshot appears in the report. How to resolve this? 
Although I am able to see the screenshots in the allure-reports > data > attachments.

**Comment by portyu9** [comment]
Hi @VarunJain1402 

Since all of the screenshot files exist under the Allure attachments directory, the screenshot **capture** is working. The next thing to verify is whether every screenshot is attached to the corresponding Allure test result.

I would make the attachment explicit in `afterTest`:

```js
import allureReporter from '@wdio/allure-reporter'

afterTest: async function (test, context, { error }) {
    if (!error) {
        return
    }

    const screenshot = await browser.takeScreenshot()

    const capability =
        `${browser.capabilities.browserName} ` +
        `${browser.capabilities.browserVersion ?? ''}`

    allureReporter.addAttachment(
        `Failure screenshot - ${capability}`,
        Buffer.from(screenshot, 'base64'),
        'image/png'
    )
}
```

Each capability runs in its own WDIO worker, so `afterTest` should execute independently for each one.

If the report still displays only one attachment, inspect the generated files in **`allure-results` before running `allure generate`**. Each capability's `*-result.json` should contain an attachment entry referring to its screenshot.

If all four result JSON files contain their own attachment references but the generated UI shows only one, that points to the Allure report-generation/grouping side rather than `browser.takeScreenshot()`.

There isn't enough configuration/result JSON in the discussion to identify that final step conclusively.

I reviewed the whole thread; there were no screenshots, reproduction links, or replies.

Please mark this answer as accepted if it helped, thank you.
